plugins/rildev.c

```diff
@@ -102,6 +108,172 @@ static int create_rilmodem(const char *ril_type, int slot)
 	return 0;
 }
 
+static int probe_rilmodem(const char *ril_type, int slot)
+{
+	char *socket;
+	int retval;
+
+	/* AOSP has socket path "rild", "rild2"..., while others may differ */
+	if (slot != 0)
+		if (strcmp("qcom_msim", ril_type) == 0)
+			socket = g_strdup_printf(RIL_SOCKET_FOLDER "rild%d",
+								slot);
+		else
+			socket = g_strdup_printf(RIL_SOCKET_FOLDER "rild%d",
+								slot + 1);
+	else
+		socket = g_strdup(RIL_SOCKET_FOLDER "rild");
+
+	retval = access(socket, R_OK | W_OK);
+	if (retval < 0) {
+		if (errno == ENOENT) {
+			struct rilmodem_info *info = g_malloc0(sizeof(*info));
+
+			info->ril_type = ril_type;
+			info->slot = slot;
+			info->socket = socket;
+			g_socket_list = g_slist_prepend(g_socket_list, info);
+
+			goto exit;
+		} else {
+			ofono_info("Error probing %s: %s",
+						socket, strerror(errno));
+			g_free(socket);
+			goto exit;
+		}
+	}
+
+	ofono_info("Creating rilmodem with socket %s", socket);
+	retval = create_rilmodem(ril_type, slot, socket);
+	g_free(socket);
+
+exit:
+	return retval;
+}
+
+static void destroy_rilinfo(gpointer data)
+{
+	struct rilmodem_info *info = data;
+
+	g_free(info->socket);
+	g_free(info);
+}
+
+static gboolean inotify_event(GIOChannel *channel, GIOCondition cond,
+								gpointer data)
+{
+	char buf[4096] __attribute__((aligned(4)));
```

Comment:
The example code in the inotify man page uses __alignof(struct inotify_event) as opposed to hard-coding the 4.  This might make easier to understand.